js/ui/accordion.js

```diff
@@ -391,6 +391,9 @@ const Accordion = CollectionWidget.inherit({
                 if(this._tryParseItemPropertyName(args.fullName) === 'visible') {
                     this._updateItemHeightsWrapper(true);
                 }
+                if(this.option('repaintChangesOnly') === true && args.fullName === 'items') {
+                    this._updateItemHeightsWrapper(true);
+                }
```

Comment:
Yes, code should handle height.
And, it should handle the 'opened/closed/selected' classes and 'aria-hidden/aria-selected' attributes.

![image](https://user-images.githubusercontent.com/2094015/118993490-3a53d800-b98e-11eb-9422-e9e12c726262.png)

See https://codepen.io/DanIgnatov/pen/KKWazmz?editors=1010

Test:

        function checkItems_T992552($accordion, assert) {
            const $items = $accordion.find(`.${ACCORDION_ITEM_CLASS}`);
            assert.strictEqual($items.length, 2, '$items.length');

            assert.roughEqual($items.eq(0).outerHeight(), 93, 1.001, 'items(0) has valid height');
            assert.strictEqual($items.eq(0).is('.' + [ACCORDION_ITEM_OPENED_CLASS, SELECTED_ITEM_CLASS].join('.')), true, 'items(0) should have each of these classes');
            assert.strictEqual(!$items.eq(0).is('.' + [HIDDEN_CLASS, ACCORDION_ITEM_CLOSED_CLASS].join(', .')), true, 'items(0) should not have no one of these classes');
            assert.strictEqual($items.eq(0).attr('aria-selected'), 'true', 'items(0) should have aria-selected=true');

            assert.roughEqual($items.eq(1).outerHeight(), 43, 1.001, 'items(1) has valid height');
            assert.strictEqual($items.eq(1).is('.' + [ACCORDION_ITEM_CLOSED_CLASS].join('.')), true, 'items(1) should have each of these classes');
            assert.strictEqual(!$items.eq(1).is('.' + [HIDDEN_CLASS, ACCORDION_ITEM_OPENED_CLASS, SELECTED_ITEM_CLASS].join(', .')), true, 'items(1) should not have no one of these classes');
            assert.strictEqual($items.eq(1).attr('aria-selected'), 'false', 'items(1) should have aria-selected=false');

            const $bodyItems = $accordion.find(`.${ACCORDION_ITEM_BODY_CLASS}`);
            assert.strictEqual($bodyItems.eq(0).attr('aria-hidden'), 'false', 'bodyItems(0) should have aria-hidden=false');

            if(config.deferRendering) {
                assert.strictEqual($bodyItems.length, 1, '$bodyItems.length');
            } else {
                assert.strictEqual($bodyItems.length, 2, '$bodyItems.length');
                assert.strictEqual($bodyItems.eq(1).attr('aria-hidden'), 'true', 'bodyItems(1) should have aria-hidden=true');
            }
        }

        QUnit.test(config.message + 'accordion.selectedItem=item1 (T992552)', function(assert) {
            const items = [ { title: 'item0', text: 'Any text' }, { title: 'item1', text: 'Any text' } ];
            const $element = this.$element.dxAccordion(extend(config, { items, selectedItems: [items[0]] }));
            checkItems_T992552($element, assert);
        });

        QUnit.test(config.message + 'accordion.selectedItem=item1 -> accordion.items=[item1, newItem2] (T992552)', function(assert) {
            const items = [ { title: 'item0', text: 'Any text' }, { title: 'item1', text: 'Any text' } ];
            const $element = this.$element.dxAccordion(extend(config, { items, selectedItems: [items[0]] }));
            const instance = $element.dxAccordion('instance');

            items[1] = { title: 'new title', text: 'new text' };
            instance.option('items', items);

            checkItems_T992552($element, assert);
        });

Reply:
Yep, fixed